internal/wasm/call_context_test.go

```diff
@@ -288,3 +290,75 @@ func TestCallContext_CallDynamic(t *testing.T) {
 		require.False(t, ok, "expected no opened files")
 	})
 }
+
+func TestCallContext_SetExitCodeOnCanceledOrTimeout(t *testing.T) {
+	t.Run("timeout", func(t *testing.T) {
```

Comment:
main thing I'm concerned with is wrapped contexts, e.g. a WithValue over a Timeout or a WithCancel over a WithCancel. Mind adding tests and if there are any usage constraints as a result update the docs?

Reply:
in the implementation we are just doing <-context.Done(), and nothing above that. Wrapping context's Done is working is already ensured by Go's standard library test itests. Do you still want me to do that?

Reply:
https://github.com/tetratelabs/wazero/pull/1108/commits/6f360688011d008542d2a3649896fd8d25b666af added in any way!

Reply:
the main thinking was to prevent a change in impl breaking things. literally I was thinking if later someone decided to optimize things for fields etc. I think that was over-thinking, so clicking resolve. no action needed!